test/functional/api/v2/hosts_controller_test.rb

```diff
@@ -87,6 +88,34 @@ class HostsControllerTest < ActionController::TestCase
         assert_response :success
         assert_equal assigns('host').ansible_roles, [@ansible_role2, @ansible_role1]
       end
+
+      test 'should append a role to a host with a correct ordering' do
+        host = FactoryBot.create(:host,
+                                 :managed => false,
+                                 :ansible_role_ids => [@ansible_role3.id])
+        post :add_ansible_roles,
+             :params => {
+               :id => host.id,
+               :ansible_role_ids => [@ansible_role2.id, @ansible_role1.id]
+             },
+             :session => set_session_user
+        assert_response :success
+        assert_equal assigns('host').ansible_roles, [@ansible_role3, @ansible_role2, @ansible_role1]
+      end
+
+      test 'should remove only specified roles from a host' do
+        host = FactoryBot.create(:host,
+                                 :managed => false,
+                                 :ansible_role_ids => [@ansible_role3.id, @ansible_role2.id, @ansible_role1.id])
+        post :remove_ansible_roles,
```

Comment:
The only thing I couldn't figure out yet is how the tests are passing even though it is a post request (same for the host). Should it be a delete request?

Reply:
That's a good question... :/

Reply:
I think@adamruzicka describes it pretty well [here](https://matrix.to/#/!dxfWqeLNsImROheGXo:matrix.org/$16770832718951BGFWd:libera.chat?via=libera.chat&via=matrix.org&via=ctrl-c.liu.se)